src/main/java/seedu/address/model/assignment/Priority.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.model.assignment;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Priority Tag in ProductiveNUS.
+ */
+public class Priority {
+    public static final String LOW_PRIORITY = "LOW";
+    public static final String MEDIUM_PRIORITY = "MEDIUM";
+    public static final String HIGH_PRIORITY = "HIGH";
+    public static final String MESSAGE_CONSTRAINTS = "Priority levels should be either LOW, MEDIUM or HIGH.";
+    public enum Level {
+        NONE, LOW, HIGH, MEDIUM
+    }
+
+    public final Level priority;
+
+    /**
+     * Constructs a false Priority.
+     */
+    public Priority() {
+        priority = Level.NONE;
+    }
```

Comment:
False priority is a little misleading as it makes it sound as if the user used the wrong input format for the Priority command instead of setting no priority to the assignment. You may want to consider changing the phrasing of the JavaDocs for the constructor such that it better reflects what the constructor does :)

Reply:
Changed to "None"!